saleor/graphql/notifications/mutations/external_notification_trigger.py

```diff
@@ -0,0 +1,142 @@
+
+from ....core.permissions import AccountPermissions
+from ....graphql.utils import resolve_global_ids_to_primary_keys
+from ....webhook.payloads import (
+    generate_checkout_payload,
+    generate_customer_payload,
+    generate_fulfillment_lines_payload,
+    generate_invoice_payload,
+    generate_order_payload,
+    generate_product_payload,
+    generate_product_variant_payload,
+)
+from ...core.mutations import BaseMutation
+from ..inputs import ExternalNotificationTriggerInput
+
+
+class ExternalNotificationTrigger(BaseMutation):
+    PK__IN = "pk__in"
+    PAYLOAD_MAPPING = {
+        "Fulfillment": (
+            Fulfillment,
+            generate_fulfillment_lines_payload,
+            None,
+            PK__IN,
+            OrderPermissions.MANAGE_ORDERS,
+        ),
+        "ProductVariant": (
+            ProductVariant,
+            generate_product_variant_payload,
+            list,
+            PK__IN,
+            ProductPermissions.MANAGE_PRODUCTS,
+        ),
+        "Product": (
+            Product,
+            generate_product_payload,
+            None,
+            PK__IN,
+            ProductPermissions.MANAGE_PRODUCTS,
+        ),
+        "User": (
+            User,
+            generate_customer_payload,
+            None,
+            PK__IN,
+            AccountPermissions.MANAGE_USERS,
+        ),
+        "Checkout": (
+            Checkout,
+            generate_checkout_payload,
+            None,
+            "token__in",
+            CheckoutPermissions.MANAGE_CHECKOUTS,
+        ),
+        "Invoice": (
+            Invoice,
+            generate_invoice_payload,
+            None,
+            PK__IN,
+            OrderPermissions.MANAGE_ORDERS,
+        ),
+        "Order": (
+            Order,
+            generate_order_payload,
+            None,
+            PK__IN,
+            OrderPermissions.MANAGE_ORDERS,
+        ),
+    }
+
+    class Arguments:
+        input = ExternalNotificationTriggerInput(
+            required=True, description="Input for External Notification Trigger."
+        )
+        pluginId = graphene.String(description="The ID of notification plugin.")
+
+    class Meta:
+        description = "Send external notification to given customers."
+        error_type_class = ExternalNotificationError
+
+    @classmethod
+    def perform_mutation(cls, root, info, **data):
+        manager = info.context.plugins
+        ids, extra_payloads, external_event_type = cls.get_cleaned_data(data)
+        model_type, pks = resolve_global_ids_to_primary_keys(ids)
+        if payload_params := cls.PAYLOAD_MAPPING.get(model_type):
+            (
+                model,
+                payload_function,
+                input_type,
+                pk_lookup_type,
+                permission_type,
+            ) = payload_params
+            if cls._is_user_contain_permission(info.context, permission_type):
+                if input_type == list:
```

Comment:
Not sure what `input_type` means

Reply:
I see that it is used in mapping for productVariant. I don't think that we will need this as we're going to prepare payload for orders and customers (afair)